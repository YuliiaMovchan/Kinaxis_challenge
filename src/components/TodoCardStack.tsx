import styled from "styled-components";
import { Todo } from "../types";
import TodoCard from "./TodoCard";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 250px;
  margin: 20px;
  padding: 36px;
  background-color: #dedede;
`;

interface TodoCardStackProps {
  cards: Todo[];
  title: string;
  onStatusChange: (id: number, isComplete: boolean, note: string) => void;
  onCardEdit: (id: number, completed: boolean, note: string) => void;
}

const TodoCardStack = ({
  cards,
  title,
  onStatusChange,
  onCardEdit,
}: TodoCardStackProps) => {
  return (
    <Wrapper>
      <h2>{title}</h2>

      {cards.map((card: Todo) => (
        <TodoCard
          key={card.id}
          id={card.id}
          note={card.note}
          completed={card.completed}
          onStatusChange={onStatusChange}
          onCardEdit={onCardEdit}
        />
      ))}
    </Wrapper>
  );
};

export default TodoCardStack;
