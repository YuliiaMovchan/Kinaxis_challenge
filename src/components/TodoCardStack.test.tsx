import renderer from "react-test-renderer";
import { Todo } from "../types";
import TodoCardStack from "./TodoCardStack";

let cards: Todo[];

beforeAll(() => {
  cards = [];
  for (let i = 0; i < 4; i++) {
    cards.push({ id: i, note: `note${i}`, completed: false });
  }
});

it("renders stack", () => {
  const tree = renderer
    .create(
      <TodoCardStack
        title="title"
        cards={cards}
        onStatusChange={() => {}}
        onCardEdit={() => {}}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
