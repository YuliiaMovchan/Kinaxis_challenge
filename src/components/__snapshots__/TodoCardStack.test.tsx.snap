// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders stack 1`] = `
<div
  className="sc-jSMfEi vBoWH"
>
  <h2>
    title
  </h2>
  <div
    className="sc-bczRLJ lkLdxK"
  >
    <label
      className="sc-dkzDqf cHkOaT"
      contentEditable={false}
      htmlFor="0"
      suppressContentEditableWarning={true}
    >
      note0
    </label>
    <div
      className="sc-gsnTZi fFWGpK"
    >
      <input
        checked={false}
        className="sc-hKMtZM klVUjV"
        id="0note0"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-eCYdqJ fzDQYo"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="sc-bczRLJ lkLdxK"
  >
    <label
      className="sc-dkzDqf cHkOaT"
      contentEditable={false}
      htmlFor="1"
      suppressContentEditableWarning={true}
    >
      note1
    </label>
    <div
      className="sc-gsnTZi fFWGpK"
    >
      <input
        checked={false}
        className="sc-hKMtZM klVUjV"
        id="1note1"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-eCYdqJ fzDQYo"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="sc-bczRLJ lkLdxK"
  >
    <label
      className="sc-dkzDqf cHkOaT"
      contentEditable={false}
      htmlFor="2"
      suppressContentEditableWarning={true}
    >
      note2
    </label>
    <div
      className="sc-gsnTZi fFWGpK"
    >
      <input
        checked={false}
        className="sc-hKMtZM klVUjV"
        id="2note2"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-eCYdqJ fzDQYo"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="sc-bczRLJ lkLdxK"
  >
    <label
      className="sc-dkzDqf cHkOaT"
      contentEditable={false}
      htmlFor="3"
      suppressContentEditableWarning={true}
    >
      note3
    </label>
    <div
      className="sc-gsnTZi fFWGpK"
    >
      <input
        checked={false}
        className="sc-hKMtZM klVUjV"
        id="3note3"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-eCYdqJ fzDQYo"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
